manifests/osfamily/suse.pp

```diff
@@ -0,0 +1,66 @@
+class puppet_agent::osfamily::suse {
+  if $::operatingsystem != 'SLES' and ! $::puppet_agent::is_pe {
+    fail("${::operatingsystem} not supported")
+  }
+
+  case $::operatingsystemmajrelease {
+    '12': {
```

Comment:
Correct. I had SLES 11 in here but after further testing, I was running into issues with upgrades: 

```
Error: Execution of '/usr/bin/zypper --quiet install --auto-agree-with-licenses --no-confirm --name puppet-agent' returned 4: The following NEW package is going to be installed:
  puppet-agent
 
The following packages are going to be REMOVED:
  pe-agent pe-augeas pe-facter pe-hiera pe-libldap pe-libyaml pe-mcollective pe-mcollective-common pe-openssl pe-puppet pe-puppet-enterprise-release pe-ruby pe-ruby-augeas pe-ruby-ldap pe-ruby-rgen pe-ruby-shadow pe-ruby-stomp pe-rubygem-deep-merge pe-rubygem-net-ssh pe-virt-what
 
The following package is not supported by its vendor:
  puppet-agent
 
1 new package to install, 20 to remove.
Overall download size: 17.9 MiB. After the operation, additional 33.0 MiB will be used.
Continue? [y/n/?] (y): y
Installation of puppet-agent-1.2.0-1.sles11 failed:
(with --nodeps --force) Error: Subprocess failed. Error: RPM failed: warning: /etc/puppetlabs/mcollective/server.cfg created as /etc/puppetlabs/mcollective/server.cfg.rpmnew
warning: /etc/puppetlabs/puppet/puppet.conf created as /etc/puppetlabs/puppet/puppet.conf.rpmnew
warning: /etc/sysconfig/puppet created as /etc/sysconfig/puppet.rpmnew
insserv: script puppet: service puppet already provided!
insserv: script mcollective: service mcollective already provided!
insserv: Service network is missed in the runlevels 4 to use service puppet
insserv: Service syslog is missed in the runlevels 4 to use service puppet
insserv: exiting now!
/sbin/insserv failed, exit code 1
puppet                    0:off  1:off  2:off  3:off  4:off  5:off  6:off
insserv: script mcollective: service mcollective already provided!
insserv: script puppet: service puppet already provided!
insserv: Service network is missed in the runlevels 4 to use service puppet
insserv: Service syslog is missed in the runlevels 4 to use service puppet
insserv: exiting now!
/sbin/insserv failed, exit code 1
mcollective               0:off  1:off  2:off  3:off  4:off  5:off  6:off
error: %post(puppet-agent-1.2.0-1.sles11.x86_64) scriptlet failed, exit status 1
 
 
Abort, retry, ignore? [a/r/i] (a): a
Problem occured during or after installation or removal of packages:
Installation aborted by user
```

Was going to ping RE about it on monday